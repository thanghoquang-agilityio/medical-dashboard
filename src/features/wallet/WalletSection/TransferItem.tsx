import { Avatar, Text } from '@/components/ui';
import { fromDateToNow } from '@/utils';

export type TransferItemProps = {
  avatar: string;
  username: string;
  description: string;
  createdAt: string;
};
export const TransferItem = ({
  avatar,
  username,
  description,
  createdAt,
}: TransferItemProps) => {
  return (
    <div className="flex gap-4">
      <Avatar src={avatar} size="lg" className="aspect-square" />
      <div className="flex flex-1 flex-col gap-2">
        <Text
          variant="description"
          customClass="text-[14px]/5 font-bold text-primary-200"
        >
          {username}
        </Text>

        <Text size="xs" variant="description">
          {description}
        </Text>
      </div>
      <Text size="xs" variant="description" customClass="font-bold my-auto">
        {fromDateToNow(createdAt)}
      </Text>
    </div>
  );
};
