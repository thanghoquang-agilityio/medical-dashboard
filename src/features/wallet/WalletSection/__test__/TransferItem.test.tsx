import { render } from '@testing-library/react';
import { TransferItem } from '../TransferItem';

describe('TransferItem', () => {
  const setup = () => render(<TransferItem />);

  it('should render correctly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
