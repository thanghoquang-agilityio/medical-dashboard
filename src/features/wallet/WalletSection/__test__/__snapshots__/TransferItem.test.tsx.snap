// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferItem should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex gap-4"
  >
    <span
      class="overflow-hidden outline-none rounded-full ring-offset-background dark:ring-offset-background-dark shrink-0 min-w-12 w-12 h-12 bg-undefined ring-offset-0 ring-0 false"
      data-testid="avatar-container"
      tabindex="-1"
    >
      <img
        alt=""
        class="aspect-square"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="48"
        src="http://localhost/_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=96&q=75"
        srcset="/_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=48&q=75 1x, /_next/image?url=%2Fimages%2Favatar-thumbnail.webp&w=96&q=75 2x"
        style="color: transparent; object-fit: cover; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
        width="48"
      />
    </span>
    <div
      class="flex flex-1 flex-col gap-2"
    >
      <p
        class="text-[14px]/5 font-bold text-primary-200"
      />
      <p
        class="text-primary-200 font-medium text-xs"
      />
    </div>
    <p
      class="text-primary-200 text-xs font-bold my-auto"
    >
      a few seconds ago
    </p>
  </div>
</DocumentFragment>
`;
